tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,213 @@
+# Copyright © 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Live migration test for source host
+#
+# - Get resource data and dst host IP address
+# - Collect guest data, name, version, arch, image path, reusability ...
+# - Generate profile from autoyast template based on guest data
+# - First stage guest install
+# - Second stage guest install
+# - Get IP address of Guest and web page content. installation complete
+# - Set up passwordless ssh authentication
+# - Execute migration test cases
+# - Heath check guest on the dts host after migration
+# - clean up
+#
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package live_migration_src;
+use strict;
+use warnings;
+use base 'opensusebasetest';
+use testapi;
+use File::Copy 'copy';
+use File::Path 'make_path';
+use lockapi;
+use mmapi;
+use scheduler 'get_test_suite_data';
+
+my $vms            = {};
+my $vms_to_install = 0;
+my $images_path    = "/var/lib/libvirt/images";
+my $nfs_path       = get_var("NFS_PATH") ? get_var("NFS_PATH") : get_var('SUT_IP') . ":$images_path";
+my $guest          = get_required_var('GUEST');
+my $virt_insatll = "/usr/bin/virt-install --noautoconsole --network bridge=br0 --vcpus 2,maxvcpus=4 --memory 2048,maxmemory=4096 --events on_reboot=restart --video=vga --serial pty -x 'quiet textmode=1 sshd=1 sshpassword=novell'";
```

Comment:
sshpassword is used by linuxrc not openQA

Reply:
Thank you, Felix!
If you look at https://confluence.suse.com/display/~tonyyuan/Live+migration+automation you will find the test approach quit different from what Pavel mentioned. The key difference is network. bridged network + autoyast init + after.local make thet automation code very simple and clean.

Reply:
No - it makes it complicated, hard to read and most of all unrelated to the scope of the goal.

Reply:
'quiet textmode=1 sshd=1 sshpassword=novell'" is for debug. It could be removed.

Reply:
> No - it makes it complicated, hard to read and most of all unrelated to the scope of the goal.

Please be more specific.

Reply:
OK:
 * We already have `virt_autotest::utils::import_guest()` so please use it.
 * The new autoyast profiles are nice but they are not related to this merge request.
 * The bridged network is tested separately - please use the fixed address guests from `%virt_autotest::common::guests`.

Reply:
If we look at my automaton you will see the test modules: prepare_profile, boot_from_pxe, installation, login_console, live_migration_src. 4 of 5 are reused.  If we look the source code in "live_migration_src.pm"  a lot of API and methods used there are not written by me. What I actually did is to provide a test automation approach ( or algorithm)  which put them together.   I don't think my code has any "duplication" issue.

Why were utils::import_guest(), common::guests not used in my migration automation? it's because it uses different approach from mine. We can consider approach as path (or road). Let's call my approach path A, utils::import_guest() path B.

What's Path A and B?   Path A is bridged network, auto-generated autoyast profile, mac and dynamic IP address... All are automatic.  path B is NAT network, manually-created autoyast profile, mac, static IP address. All are manually. NAT rejects outside connection.

I actually did a lot research on Paths:  A , B, C, D ... before I started coding. I chose A. It has many advantages: simple, robust and fast if It is designed well. Now here I am:  As you can see the in the verification run: in the recording, screenshots, it works!  These VR were created two weeks while debugging. The better VR will come after code polishing is done.

Pavel, you suggest Path B. It's a tough path. I don't know where it leads. I believe you don't either.  A, B are two paths. Either we go A or B. not compatible.

You are pushing B. You are actually asking me to rewrite, almost the whole code. Maybe you are not aware of it.

Reply:
@tonyyuan1 sorry, but this is simply not true. The path B is the path supported by many years of experience and research. Your is completely new. The path B is commonly used, your is not. The generated autoyast profiles are nice - I'm not asking you to throw that away but make a separate pull request for that. We were using bridged network and switched out of that as the NAT is more reliable and we test bridged network elsewhere.

Reply:
They are autoyast profile templates, not profiles. The guests and hosts installation all depend on them. You can think of them as part of source code. Cannot be separated. Regarding path A and B my point is that: they are exclusive **in this automation scenarios**. Either A or B. I am going with A. 
Bro :), sometime it seems that we are even talking about different things. I know there are misunderstandings.  It seems I just cannot make you understand Path A. Sorry for that. Maybe we take a break.  after a week or two, you understand more about A, I know more about B.... We discuss again.